### PR TITLE
Fix bugs and add message logging.

### PR DESCRIPTION
**Pull request by Rithari**
See commits for more details on bug fixes and additions.

### CODE REVIEW

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 1]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
 
+        private async Task OnMessageUpdatedAsync(Cacheable<IMessage, ulong> cacheable, SocketMessage message, ISocketMessageChannel socketChannel)
+        {
+            var channel = socketChannel as SocketTextChannel;
+            var before = await cacheable.GetOrDownloadAsync();
+            var author = before.Author;
+
+            if (
+                before.Author == _bot.CurrentUser || message.Author == _bot.CurrentUser ||
+                before.Content == "" || message.Content == ""
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs]: Incosistent style, we usually have parentheses on the same line as the conditions

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 2]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
+            _bot.MessageReceived += OnMessageReceived;
+            _bot.MessageDeleted += OnMessageRemoved;
+            _bot.MessageUpdated += OnMessageUpdatedAsync;
             _commands.CommandExecuted += OnCommandExecutedAsync;
             await _commands.AddModulesAsync(Assembly.GetEntryAssembly(), _services);
         }
 
+        private async Task OnMessageUpdatedAsync(Cacheable<IMessage, ulong> cacheable, SocketMessage message, ISocketMessageChannel socketChannel)
+        {
+            var channel = socketChannel as SocketTextChannel;
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs]: You don't check for null, using as is pointless; Replace with direct cast or `if (socketChannel is SocketTextChannel channel)`

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 3]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
+                before.Content == "" || message.Content == ""
+               )
+                return;
+
+            var embed = new EmbedBuilder
+            {
+                Color = Color.Blue,
+                Description = $":bulb: Message from '{author.Username}' edited in {channel.Mention}",
+                Fields = {
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":x: Original message: ", Value = before.Content },
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs]: Wrap lines in order to make content more readable

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 4]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
+            var before = await cacheable.GetOrDownloadAsync();
+            var author = before.Author;
+
+            if (
+                before.Author == _bot.CurrentUser || message.Author == _bot.CurrentUser ||
+                before.Content == "" || message.Content == ""
+               )
+                return;
+
+            var embed = new EmbedBuilder
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs:51]: We don't use object initializers for any other EmbedBuilder, while I do think this is readable we might want to apply it on every other instance

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 5]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
+            {
+                Color = Color.Blue,
+                Description = $":bulb: Message from '{author.Username}' edited in {channel.Mention}",
+                Fields = {
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":x: Original message: ", Value = before.Content },
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":pencil2: New message: ", Value = message.Content }
+                },
+                ThumbnailUrl = author.GetAvatarUrl(),
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs:69]: We often use Extension.RealAvatarUrl even tho I don't know if it really makes any difference

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 6]
```diff
@@ -32,11 +32,60 @@ public CommandHandler(DiscordSocketClient bot, CommandService commands, IService
...
+                Description = $":bulb: Message from '{cacheable.Value.Author.Username}' removed in {channel.Mention}",
+                Fields = {
+                    new EmbedFieldBuilder() { IsInline = false, Name = $":x: Original message: ", Value = cacheable.Value.Content },
+                },
+                ThumbnailUrl = cacheable.Value.Author.GetAvatarUrl(),
+                Timestamp = DateTime.Now
+            };
+
+            await channel.Guild.GetTextChannel(Global.Channels["modlog"]).SendMessageAsync(embed: embed.Build());
+
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs:92]: Remove spacing before brackets

File: src/OnePlusBot/Base/CommandHandler.cs [hunk 7]
```diff
@@ -98,7 +147,12 @@ private static bool IsValidReferralMessage(SocketMessage message)
 
         private static async Task OnMessageReceived(SocketMessage message)
         {
+            if (Regex.IsMatch(message.Content, @"discord(?:\.gg|app\.com\/invite)\/([\w\-]+)") && message.Channel.Id != Global.Channels["referralcodes"])
+                await message.DeleteAsync();
+
```
Comment (novelhawk) [inline src/OnePlusBot/Base/CommandHandler.cs:158]: Make sure spaces are at kept at a maximum of 2 or, only if really needed, 3

**Review by novelhawk — CHANGES_REQUESTED** [review]

File: src/OnePlusBot/Base/CommandHandler.cs (hunk 4 above)
Comment (Rithari) [reply src/OnePlusBot/Base/CommandHandler.cs:51]: Will be considered in a future update.

File: src/OnePlusBot/Base/CommandHandler.cs (hunk 5 above)
Comment (Rithari) [reply src/OnePlusBot/Base/CommandHandler.cs:69]: Should be tested in the reports, might fix the invisible avatar.

**Pull request merged**